internal/handlers/commonparams/extract_params.go

```diff
@@ -29,18 +29,42 @@ import (
 	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
 )
 
+// ErrFieldNotPopulated is returned when a required field is not populated.
+var ErrFieldNotPopulated = errors.New("required field is not populated")
+
 // ExtractParams fill passed value structure with parameters from the document.
 // If the passed value is not a pointer to the structure it panics.
 // Parameters are extracted by the field name or by the `ferretdb` tag.
 //
 // Possible tags:
-//   - `opt` - field is optional;
-//   - `non-default` - field is unimplemented for non-default values;
-//   - `unimplemented` - field is not implemented yet;
-//   - `ignored` - field is ignored.
+//   - `opt` - field is optional, the field value would not be set if it's not present in the document;
+//   - `unimplemented-non-default` - error would be returned if non-default value for the field is provided;
+//   - `unimplemented` - error would be returned if the value is present in the document;
+//   - `ignored` - field is ignored and would not be set, but it would be logged;
+//   - `positiveNumber` - provided value must be of types [int, long, double] and greater than 0,
+//     double values would be rounded to long;
+//   - `wholePositiveNumber` - provided value must be of types [int, long] and greater than 0;
+//   - `numericBool` - provided value must be of types [bool, int, long, double] and would be converted to bool;
```

Comment:
Are there examples where `numericBool` and `numericAsBool` used? I don't see them being provided as tags.